src/config.rs

```diff
@@ -61,8 +61,96 @@ struct Root {
     clouds: Clouds,
 }
 
-fn find_config() -> Option<PathBuf> {
-    let current = Path::new("./clouds.yaml");
+/// Merge two nested serde_yaml::Mapping structs
+/// The values from src are merged into dest. Values in src override values in dest.
+fn merge_mappings(src: &serde_yaml::Mapping, dest: &mut serde_yaml::Mapping) {
+    for (src_key, src_value) in src.iter() {
+        if src_value.is_mapping()
+            && dest.contains_key(src_key)
+            && dest.get(src_key).unwrap().is_mapping()
```

Comment:
Please use `if let` instead of check-then-unwrap:
```rust
if let Some(dest_value) = dest.get(src_key)
```

(maybe you need `Some(ref dest_value)`, I don't remember 100% if it's still required)